examples/step-79/step-79.cc

```diff
@@ -0,0 +1,2756 @@
+
+    void run();
+
+  private:
+    void create_triangulation();
+
+    void setup_boundary_values();
+
+    void setup_block_system();
+
+    void setup_filter_matrix();
+
+    void assemble_system();
+
+    BlockVector<double> solve();
+
+    std::pair<double, double>
+    calculate_max_step_size(const BlockVector<double> &state,
+                            const BlockVector<double> &step) const;
+
+    BlockVector<double>
+    calculate_test_rhs(const BlockVector<double> &test_solution) const;
+
+    double calculate_exact_merit(const BlockVector<double> &test_solution);
+
+    BlockVector<double> find_max_step();
+
+    BlockVector<double> compute_scaled_step(const BlockVector<double> &state,
+                                            const BlockVector<double> &step,
+                                            const double descent_requirement);
+
+    bool check_convergence(const BlockVector<double> &state);
+
+    void output_results(const unsigned int j) const;
+
+    void write_as_stl();
+
+
+    // Most of the member variables are also standard. There are,
+    // however, a number of variables that are specifically related
+    // to the optimization algorithm (such the various scalar
+    // factors below) as well as the filter matrix to ensure that
+    // the design remains smooth.
+    Triangulation<dim>        triangulation;
+    FESystem<dim>             fe;
+    DoFHandler<dim>           dof_handler;
+    AffineConstraints<double> constraints;
+
+    std::map<types::global_dof_index, double> boundary_values;
+
+    BlockSparsityPattern      sparsity_pattern;
+    BlockSparseMatrix<double> system_matrix;
+
+    SparsityPattern      filter_sparsity_pattern;
+    SparseMatrix<double> filter_matrix;
+
+    BlockVector<double> system_rhs;
+    BlockVector<double> nonlinear_solution;
+
+    const double density_ratio;
+    const double density_penalty_exponent;
+    const double filter_r;
+    double       penalty_multiplier;
+    double       barrier_size;
+
+
+    TimerOutput timer;
+  };
+
+
+  // @sect3{Constructor and set up functions}
+
+
+  // This problem has quite a lot of variables. We initialize a
+  // FESystem composed of 2×dim `FE_Q(1)` elements for the
+  // displacement variable and its Lagrange multiplier, and 7
+  // `FE_DGQ(0)` elements.  These piecewise constant functions are
+  // for density-related variables: the density itself, the
+  // unfiltered density, the slack variables for the lower and upper
+  // bounds on the unfiltered density, and then Lagrange multipliers
+  // for the connection between filtered and unfiltered densities as
+  // well as for the inequality constraints.
+  //
+  // The order in which these elements appear is documented above.
+  template <int dim>
+  SANDTopOpt<dim>::SANDTopOpt()
+    : fe(FE_DGQ<dim>(0) ^ 1,
+         (FESystem<dim>(FE_Q<dim>(1) ^ dim)) ^ 1,
+         FE_DGQ<dim>(0) ^ 1,
+         (FESystem<dim>(FE_Q<dim>(1) ^ dim)) ^ 1,
+         FE_DGQ<dim>(0) ^ 5)
```

Comment:
Is  `^ 1` needed?

Reply:
They are not actually needed, but I like them - they make it look more uniform. 
I can of course change it if that would make it easier to read or if that would be normal practice.

Reply:
Hm, I learned this syntax from looking at this program the first time around and changed all of the other tutorials to it. But that didn't find universal approval, see #11066. Would you mind switching to the old-style syntax?

Reply:
Easy enough!